language/resource-viewer/src/resolver.rs

```diff
@@ -0,0 +1,194 @@
+use crate::module_cache::ModuleCache;
+use anyhow::{anyhow, Result};
+use libra_state_view::StateView;
+use libra_types::{
+    access_path::AccessPath,
+    account_address::AccountAddress,
+    language_storage::{ModuleId, StructTag, TypeTag},
+};
+use move_core_types::identifier::{IdentStr, Identifier};
+use move_vm_types::loaded_data::types::{FatStructType, FatType};
+use std::rc::Rc;
+use stdlib::{stdlib_modules, StdLibOptions};
+use vm::{
+    access::ModuleAccess,
+    file_format::{
+        SignatureToken, StructDefinitionIndex, StructFieldInformation, StructHandleIndex,
+    },
+    CompiledModule,
+};
+
+pub(crate) struct Resolver<'a> {
+    state: &'a dyn StateView,
```

Comment:
Do we need this to be `dyn`? IMO we only need this for something like heterogeneous collections...

Reply:
I'm fine with it being `dyn` right now, but generally speaking I'm doubtful about this practice, even in the VM. Do we have more than 1 instance of StateView to process in any of our code paths?